pkg/user/repository.go

```diff
@@ -18,21 +19,14 @@ type repository struct {
 	db *gorm.DB
 }
 
-type duplicateError struct{ error }
-
-func (e duplicateError) Duplicate() {}
-
-func (e duplicateError) Unwrap() error {
-	return e.error
-}
-
 func (r repository) create(u *model.User) error {
 	err := r.db.Create(&u).Error
 
 	var perr *pgconn.PgError
 	const uniqueKeyConstraint = "23505"
 	if errors.As(err, &perr) && perr.Code == uniqueKeyConstraint {
-		return duplicateError{error: fmt.Errorf("user already exists: %v", err)}
+		err := fmt.Errorf("user already exists: %v", err)
```

Comment:
Could be done in a follow up change so we get these changes in 😋 

But wouldn't it be a more helpful/cleaner message if we would do

`err := fmt.Errorf("user %q already exists", u.userName)`

(Note: this code might not compile, I did not check what field the username is in)